app_feup/lib/view/Widgets/generic_expansion_card.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:expansion_tile_card/expansion_tile_card.dart';
+
+abstract class GenericExpansionCard extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() {
+    return GenericExpansionCardState();
+  }
+
+  String getTitle();
+  Widget buildCardContent(BuildContext context);
+}
+
+class GenericExpansionCardState extends State<GenericExpansionCard> {
+  final GlobalKey<ExpansionTileCardState> cardA = GlobalKey();
+  final double padding = 12.0;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+        padding: EdgeInsets.symmetric(vertical: 10, horizontal: 20),
```

Comment:
A small change to make it more like the generic_card file

```suggestion
    return Container(
        margin: EdgeInsets.symmetric(vertical: 10, horizontal: 20),
```